doc/source/dev/contributor/benchmarking.rst

```diff
@@ -0,0 +1,210 @@
+
+All of the commands above display the results in plain text in the
+console, and the results are not saved for comparison with future
+commits. For greater control, a graphical view, and to have results
+saved for future comparison, you can use |run-py|_
+to run the benchmarks.
+
+``run.py`` is a "wrapper" for the ``asv`` terminal command, the use of
+which is described in `Using airspeed velocity`_. ``run.py`` simply sets
+some environment variables for you, then sends all remaining command
+line arguments to ``asv``.
+
+To use it, navigate to ``scipy/benchmarks`` in the console and then
+execute:
+
+::
+
+   python run.py run
+
+Just like ``asv run`` from the asv documentation, this command runs the
+whole benchmark suite and saves the results for comparison against
+future commits.
+
+To run only a single benchmark, such as ``KleeMinty`` from
+``optimize_linprog.py``:
+
+::
+
+   python run.py run --bench optimize_linprog.KleeMinty
+
+(Note that the ``--bench`` option here is sent to ``asv``; its meaning
+is different than the ``--bench`` option for ``runtests.py``.)
+
+One great feature of ``asv`` is that it can automatically run a
+benchmark not just for the current commit, but for every commit in a
+range. ``linprog`` ``method='interior-point'`` was merged into SciPy
+with commit |7fa17f2369e0e5ad055b23cc1a5ee079f9e8ca32|_, so let’s
+run the ``KleeMinty`` benchmark for 10 commits between then and now to
+track its performance over time.
+
+::
+
+   python run.py run --bench optimize_linprog.KleeMinty --steps 10 7fa17f..
+
+.. note::
+
+   this will take a while, because SciPy has to be rebuilt for each
+   commit! For more information about specifying ranges of commits, see
+   the `git revisions documentation`_.
+
+To "publish" the results (prepare them to be viewed) and "preview" them
+in an interactive console:
+
+::
+
+   python run.py publish
+   python run.py preview
+
+ASV will report that it is running a server. Using any browser, you can
+review the results by navigating to http://127.0.0.1:8080 (local
+machine, port 8080).
+
+For much more information about the ``asv`` commands accessible via
+``run.py``, see the airspeed velocity `Commands`_ documentation. (Tip:
+check out the ``asv find`` command and the ``--quick``,
+``--skip-existing-commits``, and ``--profile`` options for ``asv run``.)
+
+.. _git revisions documentation: https://git-scm.com/docs/gitrevisions#_specifying_ranges
+.. _Commands: https://asv.readthedocs.io/en/stable/commands.html#commands
+.. _airspeed velocity: https://github.com/airspeed-velocity/asv
+.. _Using airspeed velocity: https://asv.readthedocs.io/en/stable/using.html#running-benchmarks
+.. _Benchmarking SciPy: https://youtu.be/edLQ8KRpupQ
+.. _airspeed velocity (asv) documentation: https://asv.readthedocs.io/en/stable/
+.. _airspeed velocity of an unladen scipy: https://pv.github.io/scipy-bench/
+.. _Writing benchmarks: https://asv.readthedocs.io/en/stable/writing_benchmarks.html
```

Comment:
As if working with Sphinx links wasn't fun enough already.

Reply:
Added to `conf.py`
```
intersphinx_mapping = {
    'python': ('https://docs.python.org/dev', None),
    'numpy': ('https://docs.scipy.org/doc/numpy', None),
    'matplotlib': ('https://matplotlib.org', None),
    'asv': ('https://asv.readthedocs.io/', None),
}
```
Added to `benchmarking.rst`:
```
:ref:`Writing benchmarks <asv:writing-benchmarks>`
```
Didn't work. By repeatedly editing whitespace elsewhere in the file it will probably fix itself eventually, but hasn't yet.

Reply:
`https://asv.readthedocs.io/en/stable/` FTW.

Reply:
Did the same for A Guide to NumPy/SciPy Documentation.
Tried to do the same for NumPy Distutils - Users Guide and NumPy/SciPy Testing Guidelines in the Contributor Guide TOC, but those documents don't have reference labels.